src/receivers/AwsLambdaReceiver.ts

```diff
@@ -0,0 +1,245 @@
+import { Logger, ConsoleLogger, LogLevel } from '@slack/logger';
+import querystring from 'querystring';
+import crypto from 'crypto';
+import tsscmp from 'tsscmp';
+import App from '../App';
+import { Receiver, ReceiverEvent } from '../types/receiver';
+import { ReceiverMultipleAckError } from '../errors';
+
+export interface AwsEvent {
+  body: string | null;
+  headers: any;
+  multiValueHeaders: any;
+  httpMethod: string;
+  isBase64Encoded: boolean;
+  path: string;
+  pathParameters: any | null;
+  queryStringParameters: any | null;
+  multiValueQueryStringParameters: any | null;
+  stageVariables: any | null;
+  requestContext: any;
+  resource: string;
+}
+
+export type AwsCallback = (error?: Error | string | null, result?: any) => void;
+
+export interface AwsResponse {
+  statusCode: number;
+  headers?: {
+    [header: string]: boolean | number | string;
+  };
+  multiValueHeaders?: {
+    [header: string]: Array<boolean | number | string>;
+  };
+  body: string;
+  isBase64Encoded?: boolean;
+}
+
+export type AwsHander = (event: AwsEvent, context: any, callback: AwsCallback) => Promise<AwsResponse>;
+
+export interface AwsLambdaReceiverOptions {
+  signingSecret: string;
+  logger?: Logger;
+  logLevel?: LogLevel;
+}
+
+export interface AwsLambdaReceiverInstallerOptions {
+  installPath?: string;
+  redirectUriPath?: string;
+}
+
+export default class AwsLambdaReceiver implements Receiver {
+  private signingSecret: string;
+
+  private app?: App;
+
+  private logger: Logger;
+
+  constructor({ signingSecret, logger = undefined, logLevel = LogLevel.INFO }: AwsLambdaReceiverOptions) {
+    // Initialize instance variables, substituting defaults for each value
+    this.signingSecret = signingSecret;
+    this.logger =
+      logger ??
+      (() => {
+        const defaultLogger = new ConsoleLogger();
+        defaultLogger.setLevel(logLevel);
+        return defaultLogger;
+      })();
+  }
+
+  public init(app: App): void {
+    this.app = app;
+  }
+
+  public start(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ..._args: any[]
+  ): Promise<AwsHander> {
+    return new Promise((resolve, reject) => {
+      try {
+        const handler = this.toHandler();
+        resolve(handler);
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  public stop(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ..._args: any[]
+  ): Promise<void> {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return new Promise((resolve, _reject) => {
+      resolve();
+    });
+  }
+
+  public toHandler(): AwsHander {
```

Comment:
Please ignore if irrelevant:
```suggestion
  public toHandler(): AwsHandler {
```